pkg/storage/stores/shipper/compactor/deletion/delete_request.go

```diff
@@ -3,22 +3,34 @@ package deletion
 import (
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/prometheus/model/labels"
-	"github.com/prometheus/prometheus/promql/parser"
 
 	"github.com/grafana/loki/pkg/storage/stores/shipper/compactor/retention"
 )
 
-// DeleteRequest holds all the details about a delete request.
 type DeleteRequest struct {
 	RequestID string              `json:"request_id"`
 	StartTime model.Time          `json:"start_time"`
 	EndTime   model.Time          `json:"end_time"`
-	Selectors []string            `json:"selectors"`
+	Queries   []string            `json:"queries"`
 	Status    DeleteRequestStatus `json:"status"`
 	CreatedAt model.Time          `json:"created_at"`
 
-	UserID   string              `json:"-"`
-	Matchers [][]*labels.Matcher `json:"-"`
+	UserID   string            `json:"-"`
+	matchers []*labels.Matcher `json:"-"`
+}
+
+func (d *DeleteRequest) AddQueries(queries []string) error {
+	d.Queries = queries
+	for _, query := range queries {
+		matchers, err := parseDeletionQuery(query)
+		if err != nil {
+			return err
+		}
+
+		d.matchers = append(d.matchers, matchers...)
```

Comment:
Wouldn't merging the matchers result in incorrect selection of logs? `{namespace="loki-dev"}` and `{container="compactor"}` would select totally different streams than `{namespace="loki-dev", container="compactor"}`

Reply:
This is fixed in [this pr](https://github.com/MichelHollands/loki/pull/8) that changes the API to be onle LogQL query at a time.